js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+		text = yoastShortcodePlugin.getContentTinyMCE();
+	}
+	else if ( ev.currentTarget.tagName === 'SPAN' ) {
+		text = ev.currentTarget.textContent;
+	}
+	else {
+		text = ev.currentTarget.value;
+	}
+
+	yoastShortcodePlugin.checkShortcodes( text );
+	yoastShortcodePlugin.parseShortcodes( yoastShortcodePlugin.declareReloaded );
+};
+
+/**
+ * gets content from the content field, if tinyMCE is initialized, use the getContent function to get the data from tinyMCE
+ * @returns {String}
+ */
+YoastShortcodePlugin.prototype.getContentTinyMCE = function() {
+	var val = document.getElementById( 'content' ).value;
+	if ( tinyMCE.editors.length !== 0 ) {
+		val = tinyMCE.get( 'content' ).getContent();
+	}
+
+	return val;
+};
+
+/* SHORTCODE PARSING */
+
+/**
+ * Add a shortcode to the unparsed shortcodes unless was already parsed.
+ *
+ * @param {string} shortcode
+ */
+YoastShortcodePlugin.prototype.checkShortcode = function( shortcode ) {
+	var already_exists = false;
+
+	for ( var i in this.parsedShortcodes ) {
+		if (this.parsedShortcodes[ i ].shortcode === shortcode) {
+			already_exists = true;
+		}
+	}
+
+	if ( already_exists === false ) {
+		this.unparsedShortcodes.push( shortcode );
+	}
+};
+
+/**
+ * Checks a piece of text for shortcodes and adds them to the unparsed shortcodes if they haven't been parsed yet.
+ *
+ * @param {string} text
+ * @returns {boolean}
+ */
+YoastShortcodePlugin.prototype.checkShortcodes = function( text ) {
+	if ( typeof text !== 'string') {
+		console.error( 'Failed to get unparsed shortcodes. Expected parameter to be a string, instead received' + typeof text );
+		return false;
+	}
+
+	var shortcodes = this.matchShortcodes( text );
+
+	for ( var i in shortcodes ) {
+		var shortcode = shortcodes[ i ];
+		if ( this.unparsedShortcodes.indexOf( shortcode ) === -1 ) {
+			this.checkShortcode( shortcode );
+		}
+	}
+
+	return true;
+};
+
+/**
+ * Matches a text for shortcodes.
+ *
+ * @param {string} text
+ * @returns {array} The matched shortcodes
+ */
+YoastShortcodePlugin.prototype.matchShortcodes = function( text ) {
+	var captures = this.getCapturesFromText( text );
+	for ( var i in captures ) {
+		text = text.replace(captures[ i ], '');
+	}
+	// Fetch the non capturing shortcodes from the text.
+	var nonCaptureRegex = '\\[' + this.keywordRegex + '[^\\]]*?\\]';
+	var nonCaptures = text.match(new RegExp(nonCaptureRegex, 'g')) || [];
+
+	return captures.concat( nonCaptures );
+};
+
+/**
+ * Matches the capturing shortcodes from a given piece of text, based on valid shortcode tags.
+ *
+ * @param {string} text
+ * @returns {Array}
+ */
+YoastShortcodePlugin.prototype.getCapturesFromText = function( text ) {
+	var captures = [];
+
+	// First identify which tags are being used in a capturing shortcode by looking for closing tags.
+	var closingTagRegex = '\\[\\/' + this.keywordRegex + '\\]';
+	var captureKeywords = ( text.match( new RegExp(closingTagRegex, 'g') ) || [] ).join(' ').match(new RegExp( this.keywordRegex, 'g') );
```

Comment:
It is better to cache the `RegExp`s because it can be an expensive operation to construct a regex.